File: app/src/main/java/suzp1984/github/io/exapidemo/ExApp.java
package suzp1984.github.io.exapidemo;

import android.app.Application;
import android.os.StrictMode;

import butterknife.ButterKnife;

/**
 * Created by jacobsu on 9/25/16.
 */

public class ExApp extends Application {

    public void onCreate() {
        super.onCreate();

        if (BuildConfig.DEBUG) {
            StrictMode.setThreadPolicy(new StrictMode.ThreadPolicy.Builder()
                    .detectAll()
                    .penaltyLog()
                    .penaltyDialog()
                    .build());

            StrictMode.setVmPolicy(new StrictMode.VmPolicy.Builder()
                    .detectAll()
                    .penaltyLog()
                    .build());

            ButterKnife.setDebug(true);
        }
    }
}

File: app/src/main/kotlin/suzp1984/github/io/exapidemo/ExApp.kt
package suzp1984.github.io.exapidemo

import android.app.Application
import android.os.StrictMode

import butterknife.ButterKnife

/**
 * Created by jacobsu on 9/25/16.
 */

class ExApp : Application() {

    override fun onCreate() {
        super.onCreate()

        if (BuildConfig.DEBUG) {
            StrictMode.setThreadPolicy(StrictMode.ThreadPolicy.Builder()
                    .detectAll()
                    .penaltyLog()
                    .penaltyDialog()
                    .build())

            StrictMode.setVmPolicy(StrictMode.VmPolicy.Builder()
                    .detectAll()
                    .penaltyLog()
                    .build())

            ButterKnife.setDebug(true)
        }
    }
}
